feat : 유저(고객) 생성 기능 추가

Pascal-Case opened the pull request:
### 변경사항

**AS-IS**

**TO-BE**
 - 유저 엔티티 작성
 - 유저 등록 컨트롤러, 서비스 로직 작성 및 테스트 코드 작성
 - Swagger 문서 작성

### 테스트

- [x] 테스트 코드
- [x] API 테스트

Pascal-Case committed:
docs: add pull_request_template.md

Pascal-Case committed:
docs: add user entity

- 기본 디렉토리 구조 생성
- 유저 엔티티 추가

Pascal-Case committed:
chore: 개발용 데이터베이스 연결

- 로컬 mysql 연결 설정

Pascal-Case committed:
chore: Swagger config 파일 추가

Pascal-Case committed:
feat: 유저 등록 api 작성 등

- UserJoinForm 추가
- UserController 작성
- Swagger 작성

Pascal-Case committed:
feat: 유저 등록 기능 추가

- 유저 등록 api 작성
- 관련 service, repository 작성
- JpaAuditing 관련 파일 추가
- Swagger 작성

Pascal-Case committed:
refactor: 유저 회원가입 메서드 이름 변경

Pascal-Case committed:
test: 유저 생성 테스트 코드 작성

- logger 추가
- 유저 생성 컨트롤러, 서비스 테스트 코드 작성

mingming-mentor commented on src/main/java/jyang/deliverydotdot/domain/User.java:
```diff
@@ -0,0 +1,56 @@
+
+  private String address;
+
+  private String deletedAt;
```
deletedAt인데 String으로 저장되는 이유가 있을까요? OffsetDateTime이나 LocalDateTime을 활용하셔도 좋을 것 같아요

Pascal-Case replied:
피드백 감사합니다.

mingming-mentor commented on src/main/java/jyang/deliverydotdot/service/UserService.java:
```diff
@@ -0,0 +1,52 @@
+        .authType(LOCAL)
+        .build();
+
+    validateRegisterUser(savedUser);
```
요 validateRegister가 savedUser 생성보다 먼저 일어나면 좋을 것 같아요. 왜냐하면 만약 user의 loginId가 중복되면 불필요한 savedUser instance가 만들어지게 되는 것이거든요. 가능하면 validation 로직을 메소드 앞쪽에 배치해주시면 좋을 것 같습니다

Pascal-Case replied:
불필요한 인스턴스가 생성되는지 확인하는것도 중요한거 같습니다. 피드백 감사합니다.

mingming-mentor reviewed (approved):
리뷰드렸습니다! 수고 많으셨어요

Pascal-Case committed:
fix: User 엔티티, User Service 피드백 사항 수정

- User 엔티티 deletedAt의 타입을 LocalDateTime으로 변경
- UserService의 validateRegisterUser를 saveUser 엔티티 생성 전으로 이동
